Zotero/Controllers/AttachmentCreator.swift

```diff
@@ -40,20 +40,49 @@ struct AttachmentCreator {
             return nil
         }
 
-        var attachmentData = attachmentData(for: item)
-
-        guard !attachmentData.isEmpty else { return nil }
-
-        attachmentData.sort { lData, rData in
-            mainAttachmentsAreInIncreasingOrder(lData: (lData.1, lData.3, lData.4), rData: (rData.1, rData.3, rData.4))
+        return bestFileAttachmentIfAny(for: item) ?? firstLinkedURLAttachmentIfAny(for: item, urlDetector: urlDetector)
+
+        func bestFileAttachmentIfAny(for item: RItem) -> Attachment? {
```

Comment:
This looks a bit messier than it was, while it's still pretty much just moved around without many changes. I'd keep the separation between filtering out items (previously in `attachmentData(for:)` and sorting based on priority, which is still in place (`mainAttachmentsAreInIncreasingOrder`). Maybe even add another one to create `Attachment` from the main attachment data.